src/language/mixins.md

```diff
@@ -89,6 +89,50 @@ A `mixin` declaration defines a mixin. A `class` declaration defines a [class][]
 A `mixin class` declaration defines a class that is usable as both a regular class
 and a mixin, with the same name and the same type.
 
+Any restrictions that apply to classes and mixins also apply to mixin classes:
+
+- Mixins can't have `extends` or `with` clauses, so neither can a `mixin class`.
+- Classes can't have an `on` clause, so neither can a `mixin class`. 
+
+### `abstract mixin class`
+
+You can achieve similar behavior to the `on` directive for a mixin class. 
```

Comment:
This is much better IMO and clearly explains why you might want to use this approach.

I'm not sure how convincing the example is though...

There are some examples in the wild (e.g. `MapBase` in the SDK and `BaseClient` in `package:http`) but they are too complex and don't go with the Musician theme.